mesonbuild/modules/pkgconfig.py

```diff
@@ -136,6 +138,34 @@ def generate(self, state, args, kwargs):
         pub_reqs = mesonlib.stringlistify(kwargs.get('requires', []))
         priv_reqs = mesonlib.stringlistify(kwargs.get('requires_private', []))
         conflicts = mesonlib.stringlistify(kwargs.get('conflicts', []))
+
+        def parse_variable_list(stringlist):
+            reserved = ['prefix', 'libdir', 'includedir']
+            variables = []
+            for var in stringlist:
+                l = var.split('=')
+                if len(l) < 2:
+                    raise mesonlib.MesonException('Variables must be in \'name=value\' format')
+
+                # foo=bar=baz is ('foo', 'bar=baz')
+                k, v = l[0], '='.join(l[1:])
+                if not k or not v:
+                    raise mesonlib.MesonException('Variables must be in \'name=value\' format')
+
+                # Variable names must not contain whitespaces but values
+                # may, so we can only check the first character of v
+                if any(c.isspace() for c in k) or v[0].isspace():
```

Comment:
This would mean that people can't write `key = value`, which some people might consider preferable. Maybe we should `strip()` k and v immediately after splitting?

OTOH if pkg-config requires that the space not be there (I don't know if it does) then requiring the same here is ok.